packages/synapse-react-client/src/components/DownloadCart/TableQueryActionsRequired.tsx

```diff
@@ -0,0 +1,78 @@
+import { Box, Typography } from '@mui/material'
+import {
+  ColumnModel,
+  QueryBundleRequest,
+} from '@sage-bionetworks/synapse-types'
+import { noop, times } from 'lodash-es'
+import { useEffect } from 'react'
+import { useGetActionsRequiredForTableQuery } from '../../synapse-queries/entity/useActionsRequiredForTableQuery'
+import useTrackTransientListItems from '../../utils/hooks/useTrackTransientListItems'
+import { LoadingActionRequiredCard } from './ActionRequiredCard/ActionRequiredCard'
+import { ActionRequiredListItem } from './ActionRequiredListItem'
+
+export type TableQueryActionsRequiredProps = {
+  /** The query request to use to select rows to check if actions are required. */
+  queryBundleRequest: QueryBundleRequest
+  /** The column models of the table */
+  columnModels: ColumnModel[]
+  /** Invoked when a user clicks "View Sharing Settings" for a set of files that require the Download permission*/
+  onViewSharingSettingsClicked?: (benefactorId: string) => void
+  /** Invoked when the number of required actions changes. Can be used to set UI state based on if all actions have been fulfilled */
+  onNumberOfRequiredActionsChanged?: (actionsRequired: number) => void
```

Comment:
We will use this component to fetch and display the required actions. When the set of required actions is updated, this callback is invoked. So when this is called with 0, we know all actions have been completed.